pkg/asset/installconfig/azure/validation.go

```diff
@@ -88,59 +88,66 @@ func ValidateDiskEncryptionSet(client API, ic *types.InstallConfig) field.ErrorL
 	}
 
-	if typeMeta == nil {
-		errMsg := fmt.Sprintf("not found in region %s", region)
-		return append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
+	val, ok = capabilities["MemoryGB"]
+	if ok {
+		memory, err := strconv.ParseFloat(val, 0)
+		if err != nil {
+			return append(allErrs, field.InternalError(fieldPath, err))
+		}
+		if memory < float64(req.minimumMemory) {
+			errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d GB Memory", req.minimumMemory)
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
+		}
+	} else {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, "capability not found: MemoryGB"))
 	}
 
-	ultraSSDAvailable := false
-	for _, capability := range *typeMeta.Capabilities {
+	val, ok = capabilities["PremiumIO"]
+	if diskType == "Premium_LRS" && ok {
+		if strings.EqualFold(val, "False") {
+			errMsg := fmt.Sprintf("PremiumIO not supported for instance type %s", instanceType)
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("osDisk", "diskType"), diskType, errMsg))
+		}
+	} else {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, "capability not found: PremiumIO"))
+	}
 
-		if strings.EqualFold(*capability.Name, "vCPUsAvailable") {
-			cpus, err := strconv.ParseFloat(*capability.Value, 0)
-			if err != nil {
-				return append(allErrs, field.InternalError(fieldPath, err))
-			}
-			if cpus < float64(req.minimumVCpus) {
-				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d vCPUsAvailable", req.minimumVCpus)
-				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
-			}
-		} else if strings.EqualFold(*capability.Name, "MemoryGB") {
-			memory, err := strconv.ParseFloat(*capability.Value, 0)
-			if err != nil {
-				return append(allErrs, field.InternalError(fieldPath, err))
-			}
-			if memory < float64(req.minimumMemory) {
-				errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d GB Memory", req.minimumMemory)
-				allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
-			}
-		} else if diskType == "Premium_LRS" && strings.EqualFold(*capability.Name, "PremiumIO") {
-			if strings.EqualFold(*capability.Value, "False") {
-				errMsg := fmt.Sprintf("PremiumIO not supported for instance type %s", instanceType)
-				allErrs = append(allErrs, field.Invalid(fieldPath.Child("osDisk", "diskType"), diskType, errMsg))
-			}
-		} else if strings.EqualFold(*capability.Name, "UltraSSDAvailable") {
-			ultraSSDAvailable = strings.EqualFold(*capability.Value, "True")
-		} else if strings.EqualFold(*capability.Name, string(aztypes.VMnetworkingTypeAccelerated)) {
-			if vmNetworkingType == string(aztypes.VMnetworkingTypeAccelerated) && !strings.EqualFold(*capability.Value, "True") {
+	if vmNetworkingType != "" {
```

Comment:
```suggestion
	if vmNetworkingType == string(aztypes.VMnetworkingTypeAccelerated {
```
The only thing we need to validate is that the instance type supports Accelerated network when it's set in the install-config.